fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -343,14 +358,22 @@ public void close() {
     /**
      * Run after a single transactional call within runAsync.
      */
-    private <R> Pair<R, Throwable> runAsyncPostTransaction(R result, Throwable exception) {
-        if (exception == null) {
-            successCount++;
-            if (successCount >= increaseLimitAfter && limit < maxLimit) {
-                increaseLimit();
+    @VisibleForTesting
+    <R> Pair<R, Throwable> runAsyncPostTransaction(R result, Throwable exception, AtomicLong recordsScanned) {
+        if (increaseLimitAfter > 0) {
+            if (exception == null) {
+                successCount++;
+                if (successCount >= increaseLimitAfter && limit < maxLimit) {
+                    increaseLimit();
+                }
+            } else {
+                successCount = 0;
             }
+        }
+        if (exception == null) {
+            totalRecordsScanned.addAndGet(recordsScanned.get());
         } else {
-            successCount = 0;
+            recordsScanned.set(0);
```

Comment:
I see; this is called only by `runAsync` and `buildEndpoints`, and both always use a fresh `recordsScanned` `AtomicLong`, so it won't ever reset `totalRecordsScanned` even though that gets handed off to a few of the functions within this class.